subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy

```diff
@@ -958,6 +958,8 @@ task generate(type: TransformerTask) {
             def srcDir = file('src')
             // Note: task mutates inputs _without_ declaring any inputs or outputs
             task src1 {
+                outputs.dir(srcDir)
+                outputs.upToDateWhen { false }
```

Comment:
What was the problem here? The comment above actually says the task doesn't declare any outputs.

Reply:
When using fine-grained invalidation, then the task must declare its outputs. If it doesn't, Gradle does not use the wrong state from the file system mirror.

For this PR, which does not do fine grained invalidation, shall I revert the changes to the test?

Reply:
Yes please.

Reply:
Done.